src/researches/metaDescriptionKeyword.js

```diff
@@ -1,19 +1,43 @@
-import matchTextWithWord from "../stringProcessing/matchTextWithWord.js";
-
-import { escapeRegExp } from "lodash-es";
+import matchTextWithArray from "../stringProcessing/matchTextWithArray";
+import getSentences from "../stringProcessing/getSentences";
+import escapeRegExp from "lodash-es/escapeRegExp";
 
 /**
  * Matches the keyword in the description if a description and keyword are available.
- * default is -1 if no description and/or keyword is specified
+ * Returns null if no description is specified in the given paper.
  *
  * @param {Paper} paper The paper object containing the description.
- * @returns {number} The number of matches with the keyword
+ * @param {Researcher} researcher the researcher object to gather researchers from.
+ * @returns { Object | null } The number of matches per keyword term, for the entire description and for each individual sentence.
  */
-export default function( paper ) {
+export default function( paper, researcher ) {
 	if ( paper.getDescription() === "" ) {
-		return -1;
+		return null;
 	}
-	var keyword = escapeRegExp( paper.getKeyword() );
-	return matchTextWithWord( paper.getDescription(), keyword, paper.getLocale() ).count;
+	const description = paper.getDescription();
+	const locale = paper.getLocale();
+
+	let keyPhraseForms = researcher.getResearch( "morphology" ).keyphraseForms;
```

Comment:
According to https://docs.google.com/spreadsheets/d/1eoMUVDJuBR-LYJzMkA7MjaBMvdUc39Q44npV1Orjdfc/edit?ts=5b9a2d60#gid=0 this assessment should take both keyphrase and synonyms.